File: Java/Algorithms/Series/Fibonacci.java
public class Fibonacci {
    public static void main(String[] args) {
        int n = 10; //number count in series
        int previousValue = 0;
        int nextValue = 1;

        System.out.print("Fibonacci series of " + n + " numbers: ");

        for (int i = 1; i <= n; i++) {
            System.out.print(previousValue + " ");

            int sum = previousValue + nextValue;
            //Each iteration, assign nextValue to the previousValue
            previousValue = nextValue;
            //assign sum of last two values to the next value in the series
            nextValue = sum;
        }
    }
}

File: C++/Algorithms/Dynamic Programming/Fibonacci.cpp
#include <iostream>
#include <vector>
using namespace std;
int main()
{
    int n;
    cin >> n;
    vector<int> v(n + 1, 0);
    v[1] = 1;
    for (int i = 2; i <= n; i++)
        v[i] = v[i - 1] + v[i - 2];
    cout << "Nth fibonacci number is : " << v[n] << " ";
}